src/test/java/com/ge/predix/logback/PredixEncoderTest.java

```diff
@@ -226,79 +229,126 @@ public void testPredixEncoderWithNoMessageLineSeparator() throws IOException {
+        mdc.remove(CORRELATION_KEY);
+
+        ILoggingEvent input = createLogEvent(MESSAGE_TEXT, null, null, mdc);
+
+        String expected = getExpectedOutput(input.getTimeStamp(), MESSAGE_OUTPUT, null);
+
+        String actual = encodeToPredixFormat(input, CUSTOM_CORRELATION_KEY, null);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testPredixEncoderWithMissingCustomCorrelationKey() {
+
+        Map<String, String> mdc = new HashMap<>(MDC);
+        mdc.remove(CORRELATION_KEY);
+
+        ILoggingEvent input = createLogEvent(MESSAGE_TEXT, null, null, mdc);
+
+        String expected = getExpectedOutput(input.getTimeStamp(), "", MESSAGE_OUTPUT, false, null);
+
+        String actual = encodeToPredixFormat(input, CUSTOM_CORRELATION_KEY, null);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testPredixEncoderWithMissingCustomCorrelationKeyDoesNotUseDefaultCorrelationKey() {
+
+        ILoggingEvent input = createLogEvent(MESSAGE_TEXT, null, null);
+
+        String expected = getExpectedOutput(input.getTimeStamp(), "", MESSAGE_OUTPUT, false, null);
+
+        String actual = encodeToPredixFormat(input, CUSTOM_CORRELATION_KEY, null);
         Assert.assertEquals(expected, actual);
     }
 
     private static ILoggingEvent createLogEvent(final String message, final Object[] messageArgs,
             final Throwable throwable) {
 
+        return createLogEvent(message, messageArgs, throwable, MDC);
+    }
+
+    private static ILoggingEvent createLogEvent(final String message, final Object[] messageArgs,
+            final Throwable throwable, Map<String, String> mdc) {
+
         LoggingEvent logEvent = new LoggingEvent(null, logger, Level.INFO, message, throwable, messageArgs);
         logEvent.setTimeStamp(Instant.now().toEpochMilli());
         logEvent.setThreadName(THREAD_NAME);
         logEvent.setCallerData(new StackTraceElement[] {
                 new StackTraceElement(CLASS_NAME, METHOD_NAME, NOT_AVAILABLE_FILE_NAME, NOT_AVAILABLE_LINE_NUMBER) });
-        logEvent.setMDCPropertyMap(MDC);
+        logEvent.setMDCPropertyMap(mdc);
         return logEvent;
     }
 
-    private String encodeToPredixFormat(final ILoggingEvent logEvent) throws IOException {
+    private String encodeToPredixFormat(final ILoggingEvent logEvent) {
 
-        return encodeToPredixFormat(logEvent, null);
+        return encodeToPredixFormat(logEvent, null, null);
     }
 
-    private String encodeToPredixFormat(final ILoggingEvent logEvent, final String messageLineSeparator)
-            throws IOException {
+    private String encodeToPredixFormat(final ILoggingEvent logEvent,
+            final String correlationKey, final String messageLineSeparator) {
 
         PredixEncoder<ILoggingEvent> predixEncoder = new PredixEncoder<>();
         predixEncoder.setContext(loggerContext);
+        predixEncoder.setCorrelationKey(correlationKey);
```

Comment:
You are testing correlation key as null (then default traceId is used) or custom.  How about empty string?

Reply:
Added tests for empty string and blank string.